jgrapht-core/src/main/java/org/jgrapht/alg/isomorphism/IsomorphicGraphMapping.java

```diff
@@ -44,64 +48,107 @@
      * @param core1 the mapping as array (forwards)
      * @param core2 the mapping as array (backwards)
      */
-    public IsomorphicGraphMapping(
-        GraphOrdering<V, E> g1, GraphOrdering<V, E> g2, int[] core1, int[] core2)
-    {
-        this.g1 = g1;
-        this.g2 = g2;
-        this.core1 = core1.clone();
-        this.core2 = core2.clone();
-    }
+    public IsomorphicGraphMapping(GraphOrdering<V, E> g1, GraphOrdering<V, E> g2, int[] core1, int[] core2) {
+        this.graph1 = g1.getGraph();
+        this.graph2 = g2.getGraph();
 
-    @Override
-    public V getVertexCorrespondence(V v, boolean forward)
-    {
-        GraphOrdering<V, E> firstGraph, secondGraph;
-        int[] core;
+        this.forwardMapping = new HashMap<>(this.graph1.vertexSet().size());
+        this.backwardMapping = new HashMap<>(this.graph1.vertexSet().size());
 
-        if (forward) {
-            firstGraph = g1;
-            secondGraph = g2;
-            core = core1;
-        } else {
-            firstGraph = g2;
-            secondGraph = g1;
-            core = core2;
+        for (V v: graph1.vertexSet()){
+            int vNumber = g1.getVertexNumber(v);
+            int uNumber = core1[vNumber];
+
+            if (uNumber != VF2State.NULL_NODE) {
```

Comment:
Better to introduce another

```
public static final int NULL_NODE = -1;
```
directly in the `IsomorphicGraphMapping` class.